### PR TITLE
Document custom conda environments creation

### PR DESCRIPTION
**Pull request by emileten**
I added a notebook documenting the use of `mamba` to understand and extend the base conda environment or create new ones. I put this in `platform technical documentation`, but, let me know if that's not the right place, in particular given the future reorganization that is planned. 

- Links to https://github.com/NASA-IMPACT/active-maap-sprint/issues/408

- Most of it is based on [MAAP quickstart guide (see 'Custom conda environments')](https://docs.google.com/document/d/1YSE0PcZV7N7VwCEMcay0HYxsqauqsa9Zl-DiBq-d2mA/edit#heading=h.r5sh9icr71mw). It essentially updates it with the `mamba` availability and removes the confused linked to its absence.  

- Relies on ADE workspace images available at https://ade.dit.maap-project.org, [based on the root image](https://github.com/MAAP-Project/maap-workspaces/blob/develop/base_images/vanilla/docker/Dockerfile) that has `mamba` installed. This notebook was tested on this image.

### CODE REVIEW

**Comment by emileten** [comment]
@wildintellect let me know if that at least partially addresses https://github.com/NASA-IMPACT/active-maap-sprint/issues/408. The main point here is an improved documentation in anticipation of the new images (that have mamba) being available in the production ADE. I understood these images will be based on : https://github.com/MAAP-Project/maap-workspaces/blob/develop/base_images/vanilla/docker/Dockerfile.

I am I think addressing the first but not second AC -- the second has to be addressed directly in the maap-workspace repo that has the Dockerfiles, I think.

**Comment by wildintellect** [comment]
> I am I think addressing the first but not second AC -- the second has to be addressed directly in the maap-workspace repo that has the Dockerfiles, I think.

Actually that's the point, if a new conda env is made the user has to install these libs, the workspace image will only ever have them for the `base` env.

File: docs/source/platform_technical_documentation/custom-environments.ipynb [hunk 1]
```diff
@@ -0,0 +1,168 @@
+{
```
Comment (wildintellect) [inline docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <p>Maybe just call this section Workspaces? We should probably raise a ticket that it's unclear that he Basic Stable is <code>python</code> and that all other workspaces also include <code>python</code> as part of <code>conda</code> environments.   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/159/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (wildintellect) [inline docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <ol><li>Can we include an example yaml in this repo?</li><li>Also can we show how to use this yaml to install to the <code>base</code>,installing outside of <code>base</code> is only necessary when conflicts occur.</li><li>We need another section showing how to select the correct kernel in a notebook, and that these commands could be run from a notebook too.</li></ol><p> </p><p>"Some python packages are not available via <code>mamba</code>." <code>conda-forge</code> it's critical to be clear that's where packages come from.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/159/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Comment by emileten** [comment]
@wildintellect 

> Actually that's the point, if a new conda env is made the user has to install these libs, the workspace image will only ever have them for the base env.

I added a section in the docs with a yaml file that has `maap-py` versioned in it [(it's indeed in the base environment config)](https://github.com/MAAP-Project/maap-workspaces/blob/ad5add5fe0a52f243c99e7d450c4f59944e8b617/base_images/vanilla/docker/Dockerfile#L6).
However, I do not see anywhere in the dockerfiles/start scripts `ipycmc`. I don't see it if I run `pip freeze` or `conda list` on the `base` environment. Could you point me to where it's referenced in the `base` environment setup ?

File: docs/source/platform_technical_documentation/custom-environments.ipynb (hunk 1 above)
Comment (emileten) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <p>Changed the title. </p><p> </p><blockquote>We should probably raise a ticket that it's unclear that he Basic Stable is <code>python</code> and that all other workspaces also include <code>python</code> as part of <code>conda</code> environments.</blockquote><p>You mean that the workspace titles do not reflect what they contain ? Agreed. </p>

[the remark this answers is not in the document]
Comment (emileten) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <blockquote>Somewhere we need to be clear that these modifications do not survive workspace restarts.</blockquote><p>You mean a more detailed explanation than what I have here?</p>

[the remark this answers is not in the document]
Comment (emileten) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <p>Like, explaining why these do not survive ?</p>

Comment (emileten) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <p>Thanks, agreed, will make these changes tomorrow. </p>

**Comment by abarciauskas-bgse** [comment]
@emileten Looks great to me, but I would wait for Alex M to re-review and approve

**Comment by emileten** [comment]
Thanks @abarciauskas-bgse ! Yes and I still have to address one of his comments.

File: docs/source/platform_technical_documentation/custom-environments.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (wildintellect) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <p>loose -&gt; lose</p><p> </p><p>(See next section). &lt;- add this to the end.</p>

[the remark this answers is not in the document]
Comment (wildintellect) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <p>should all examples use <code>-c conda-forge</code>? probably I don't think we want anything from <code>defaults</code>  </p>

Comment (wildintellect) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <p>still need kernel selection inside a notebook screenshot</p>

File: docs/source/platform_technical_documentation/custom-environments.ipynb [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
```
Comment (wildintellect) [inline docs/source/platform_technical_documentation/custom-environments.ipynb:1]: <p>Remove <code>basic</code> from <code>basic computing</code>   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/159/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

**Comment by emileten** [comment]
Note regarding `ipycmc` (mentioned in the issue this PR is solving). From Marjorie J Lucas (JPL) : 

> The ipycmc extension is still deployed on the (old) OPS venue, but we have not deployed it to the jupyter v3 venues yet because there is some refactoring that needs to be done for the package to work in those environments. The package is still in use, and installable like the other extensions you may have seen in the dockerfiles from maap-workspaces, but as mentioned above, there is still some work we need to get done for it to be compatible with the upgraded environments.

TL;DR `ipycmc` is pre-installed in the old images, but not yet in the new one because of compatibility issues but will be eventually. 

I will ignore `ipycmc` in this PR and postpone to a later ticket the task of mentioning it in this documentation about custom conda environments.

File: docs/source/platform_technical_documentation/custom-environments.ipynb (hunk 1 above)
Comment (emileten) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: created https://github.com/MAAP-Project/maap-documentation/issues/163

[the remark this answers is not in the document]
Comment (emileten) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: > loose -> lose
> 
> (See next section). <- add this to the end.

Done 




> should all examples use -c conda-forge? probably I don't think we want anything from defaults

Done. However, what do we not want anything from `defaults`? 

> Somewhere we need to be clear that these modifications do not survive workspace restarts. Long term they should submit tickets/coordinate with the platform team to create workspaces with all the libraries they need. This solution is to be able to experiment and test what's needed.

Made all this clearer.

File: docs/source/platform_technical_documentation/custom-environments.ipynb (hunk 2 above)
Comment (emileten) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: done

File: docs/source/platform_technical_documentation/custom-environments.ipynb (hunk 1 above)
Comment (emileten) [reply docs/source/platform_technical_documentation/custom-environments.ipynb:1]: >   Can we include an example yaml in this repo?
>     Also can we show how to use this yaml to install to the base,installing outside of base is only necessary when conflicts occur.
>     We need another section showing how to select the correct kernel in a notebook, and that these commands could be run from a notebook too.
> 
> "Some python packages are not available via mamba." conda-forge it's critical to be clear that's where packages come from.

Done

**Comment by emileten** [comment]
@wildintellect I addressed all your comments. Do you think we can merge this? 

Thanks!

**Comment by rtapella** [comment]
This is great. I am updating the Getting Started docs with content from the Quickstart guide and this is a good refresh on the conda/mamba stuff. I'll include you in review of that when it's ready to be reviewed.
